app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -554,7 +554,14 @@ private void queueDownload(Download download) {
         // So, for now we always save file to DIRECTORY_DOWNLOADS
         final String dir = Environment.DIRECTORY_DOWNLOADS;
 
-        final DownloadManager.Request request = new DownloadManager.Request(Uri.parse(download.getUrl()))
+        // block non-http/https download links
+        Uri downloadUri = Uri.parse(download.getUrl());
+        if(!TextUtils.equals(downloadUri.getScheme(), "http") && !TextUtils.equals(downloadUri.getScheme(), "https")) {
```

Comment:
Let's just use android.webkit.URLUtil.isNetworkUrl(download.getUrl()) for simplicity here.